include/picongpu/particles/collision/binary/DefaultAlg.hpp

```diff
@@ -0,0 +1,494 @@
+                particles::TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE,
+                par0
+            );
+            float_X const charge1 = picongpu::traits::attribute::getCharge(
+                particles::TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE,
+                par1
+            );
+
+            float_X const gamma0 =  picongpu::gamma< float_X >(
+                labMomentum0,
+                mass0
+          );
+            float_X const gamma1 =  picongpu::gamma< float_X >(
+                labMomentum1,
+                mass1
+            );
+
+            // [Perez 2012] (1)
+            float3_X const comsVelocity = ( labMomentum0 + labMomentum1 ) /
+                ( mass0 * gamma0 + mass1 * gamma1);
+            float_X const comsVelocityAbs2 = pmacc::math::abs2( comsVelocity );
+            float3_X comsMomentum0;
+            float_X  gammaComs, factorA, coeff0, coeff1;
+
+            if( comsVelocityAbs2 != 0.0_X )
+            {
+                gammaComs = 1.0_X / math::sqrt( 1 -
+                    comsVelocityAbs2 / ( c * c ) );
+                // used later for comsToLab:
+                factorA = ( gammaComs - 1.0_X ) / comsVelocityAbs2;
+
+                // Stared gamma times mass, from [Perez 2012].
+                coeff0 = coeff(
+                    labMomentum0,
+                    mass0,
+                    gamma0,
+                    gammaComs,
+                    comsVelocity
+                );
+                // gamma^* . mass
+                coeff1 = coeff(
+                    labMomentum1,
+                    mass1,
+                    gamma1,
+                    gammaComs,
+                    comsVelocity
+                );
+
+                // (2) in [Perez 2012]
+                comsMomentum0 = labToComs(
+                    labMomentum0,
+                    mass0,
+                    gamma0,
+                    gammaComs,
+                    factorA,
+                    comsVelocity
+                );
+            }
+            else
+            {
+                // From smilei implementation, for v_coms^2 = 0
+                gammaComs = 1.0_X;
+                // used later for comsToLab:
+                factorA =  0.5_X;
+                // Stared gamma times mass, from [Perez 2012].
+                coeff0 = mass0 * gamma0;
+                // gamma^* . mass
+                coeff1 = mass1 * gamma1;
+                comsMomentum0 = labMomentum0;
+            }
+
+
+            //  f0 * f1 * f2^2
+            //  is equal  s12 * (n12/(n1*n2)) from [Perez 2012]
+            float_X comsMomentum0Abs2 = pmacc::math::abs2( comsMomentum0 );
+
+            if( comsMomentum0Abs2 == 0.0_X &&
+                ( comsMomentum0 != float3_X{ 0.0_X, 0.0_X, 0.0_X } )) return;
+
+            float_X s12Factor0 = ( DELTA_T * coulombLog * charge0 *
+                charge0 * charge1 * charge1 ) / ( 4.0_X * PI * EPS0 * EPS0
```

Comment:
please check if PI in PIConGPU is by default `float64`, if so please use `float_X(PI)`

Reply:
For PI we already have a way to get it at any type.